### ISSUE TITLE
BBCode breaks on BSS (map) upload/update

### ISSUE THREAD

**Issue by NoffyNoffeh** [ppy/osu-web#7051]
For example on this map,

https://osu.ppy.sh/beatmapsets/1262989#osu/2625234 

I put `[profile]Aunn[/profile]` in the description

but after updating through bss, it changes to plaintext `[profile=12613340]Aunn[/profile]`. Saving without changes does not fix it. For some reason, this is not an issue when editing it via the website, even when the website adds in the user ID number. It only becomes stuck as plaintext after updating the beatmap from in-game.

**Comment by tilda** [comment]
Can confirm this is a thing. Just adding a space somewhere random and saving seems to fix it, though it does get annoying when you update a lot :p

**Comment by venix12** [comment]
actually ``[profile]Aunn[/profile]`` set on website parses to ``[profile=12613340:1337]Aunn[/profile:1337]`` (see https://github.com/ppy/osu-web/blob/master/app/Libraries/BBCodeForDB.php#L225), so my guess would be BSS sets that without a proper parsing?

possibly a thing to bring on https://github.com/ppy/osu-stable-issues instead but not really sure

**Comment by MzudemO** [comment]
This is also an issue with Unicode symbols in descriptions (e.g. ß). Updating the map turns them into the HTML representation (&szlig).

**Comment by peppy** [comment]
Just to confirm, does manually removing the `=user_id` part in BSS and saving work correctly?

**Comment by dorothy3242** [comment]
Yes, it worked when updating from BSS if I removed the `=user_id` part from the submission window description editor.

**Comment by NoffyNoffeh** [comment]
Would this be considered part of the same issue as #7219 ? Since both involve special description links breaking on BSS update. Not sure if I should leave this issue open as a separate one or not.

**Comment by peppy** [comment]
Not 100% sure on that; best leaving them separated for now.

**Comment by peppy** [comment]
Is this still an issue?

**Comment by nanaya** [comment]
Unless bbcode handling for submission server has been updated to match current osu-web then I don't think this has been fixed.

**Comment by dotremilie** [comment]
Can you rename the issue back to something with "profile" in it, as I see that searching for open issues with "profile", not finding one and creating a new issue is a common pattern?

**Comment by peppy** [comment]
Fixing this will probably require the new BSS version. I dunno I want to attempt a fix on phpbb.

**Comment by bdach** [comment]
https://osu.ppy.sh/home/changelog/cuttingedge/20251127
